app/frontend/src/features/profile/hooks/referencesHooks.ts

```diff
@@ -44,3 +61,66 @@ export function useReferencesReceived(
 
   return referencesReceivedQuery;
 }
+
+export const useListAvailableReferences = (userId: number) =>
+  useQuery<AvailableWriteReferencesRes.AsObject, GrpcError>(
+    availableWriteReferencesKey(userId),
+    () =>
+      service.references.getAvailableReferences({
+        userId,
+      })
+  );
+
+interface WriteHostRequestReferenceVariables {
+  referenceData: WriteHostRequestReferenceInput;
+}
+
+export function useWriteHostReference(userId: number) {
+  const queryClient = useQueryClient();
+  const {
+    mutate: writeHostRequestReference,
+    status,
+    reset,
+    error,
+    isLoading,
+  } = useMutation<
+    Reference.AsObject,
+    Error,
+    WriteHostRequestReferenceVariables
+  >(
+    ({ referenceData }) =>
+      service.references.writeHostRequestReference(referenceData),
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries([availableWriteReferencesKey(userId)]);
```

Comment:
We'd probably need to invalidate the query for `referencesReceivedKey` for `userId` too to ensure we can see the new references on their profile